docs: add Electron 39 blog post (#894)
electron/electronjs.org-new: blog/electron-39-0.md

In related news, [Electron Packager](https://github.com/electron/packager) v19 now enables ASAR by default. [#1841](https://github.com/electron/packager/pull/1841)

### New Features and Improvements

- Added `app.isHardwareAccelerationEnabled()`. [#48680](https://github.com/electron/electron/pull/48680)
- Added `RGBAF16` output format with scRGB HDR color space support to Offscreen Rendering. [#48504](https://github.com/electron/electron/pull/48504)
- Added `fileBacked` and `purgeable` fields to `process.getSystemMemoryInfo()` for macOS. [#47628](https://github.com/electron/electron/pull/47628) <sup>(Also in [37](https://github.com/electron/electron/pull/48143), [38](https://github.com/electron/electron/pull/48146))</sup>
- Added `tray.{get|set}AutosaveName` to enable macOS tray icons to maintain position across launches. [#47838](https://github.com/electron/electron/pull/47838) <sup>(Also in [36](https://github.com/electron/electron/pull/48078), [37](https://github.com/electron/electron/pull/48076), [38](https://github.com/electron/electron/pull/48077))</sup>

> - Added `webFrameMain.fromFrameToken(processId, frameToken)` to get a `WebFrameMain` instance from its frame token. [#47850](https://github.com/electron/electron/pull/47850) <sup>(Also in [38](https://github.com/electron/electron/pull/47942))</sup>

Reviewer: ```suggestion
```

We already mentioned this in the v38 blog post.

- Support dynamic ESM imports in non-context isolated preloads. [#48488](https://github.com/electron/electron/pull/48488) <sup>(Also in [37](https://github.com/electron/electron/pull/48487), [38](https://github.com/electron/electron/pull/48489))</sup>
- Marked the [ASAR integrity](https://www.electronjs.org/docs/latest/tutorial/asar-integrity) feature as stable. It had previously been experimental. [#48434](https://github.com/electron/electron/pull/48434)

### Breaking Changes

#### Deprecated: `--host-rules` command line switch

Chromium is deprecating the `--host-rules` switch.

You should use `--host-resolver-rules` instead.

#### Behavior Changed: window.open popups are always resizable

Per current [WHATWG spec](https://html.spec.whatwg.org/multipage/nav-history-apis.html#dom-open-dev), the `window.open` API will now always create a resizable popup window.

To restore previous behavior:

```js
webContents.setWindowOpenHandler((details) => {
  return {
    action: 'allow',
    overrideBrowserWindowOptions: {
      resizable: details.features.includes('resizable=yes'),
    },
  };
});
```

#### Behavior Changed: shared texture OSR `paint` event data structure

> When using shared texture offscreen rendering feature, the `paint` event now emits a more structured object.

Reviewer: ```suggestion
When using the shared texture offscreen rendering feature, the `paint` event now emits a more structured object.
```

It moves the `sharedTextureHandle`, `planes`, `modifier` into a unified `handle` property.

> See [here](https://www.electronjs.org/docs/latest/api/structures/offscreen-shared-texture) for more details.

Reviewer: ```suggestion
See [the `OffscreenSharedTexture` documentation](https://www.electronjs.org/docs/latest/api/structures/offscreen-shared-texture) for more details.
```

SEO :)

To restore previous behavior:

```js
webContents.setWindowOpenHandler((details) => {
  return {
    action: 'allow',
    overrideBrowserWindowOptions: {
      resizable: details.features.includes('resizable=yes'),
    },
  };
});
```

#### Behavior Changed: shared texture OSR `paint` event data structure

When using the shared texture offscreen rendering feature, the `paint` event now emits a more structured object.
It moves the `sharedTextureHandle`, `planes`, `modifier` into a unified `handle` property.
See [the `OffscreenSharedTexture` documentation](https://www.electronjs.org/docs/latest/api/structures/offscreen-shared-texture) for more details.

## End of Support for 36.x.y

Electron 36.x.y has reached end-of-support as per the project's [support policy](https://www.electronjs.org/docs/latest/tutorial/electron-timelines#version-support-policy). Developers and applications are encouraged to upgrade to a newer version of Electron.

| E39 (Oct'25) | E40 (Jan'26) | E41 (Feb'26) |
| ------------ | ------------ | ------------ |
| 39.x.y       | 40.x.y       | 41.x.y       |
| 38.x.y       | 39.x.y       | 40.x.y       |
| 37.x.y       | 38.x.y       | 39.x.y       |

> ## What's Next

Reviewer 2: consider: adding a section to mention december quiet period
